### PR TITLE
build(pipeline): tune project distribution in checking code gen

### PR DESCRIPTION
**Pull request by archerzz**
# Description

Job `Check_Code_Generation_J_R` takes much more time than others, due to the fact that it contains most project. This commit will reduce the projects distribution by moving projects with name starting with `J` to previous job and projects with name starting with `R` to next job.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by archerzz** [comment]
## Before
image


## After

image

---

Reduce 1 min for the longest job.

**Comment by m-nash** [comment]
@hallipr I know we were thinking of making this part dynamic as well with line # heuristics.  As long as this doesn't conflict with that work I think this is fine for an improvement in the meantime.

File: eng/pipelines/build.yml [hunk 1]
```diff
@@ -146,20 +146,20 @@ stages:
...
-          name: Check_Code_Generation_J_R
-          filter: "^[j-r]"
+          name: Check_Code_Generation_M
+          filter: "^m"
       - template: check-code-generation.yml
         parameters:
-          name: Check_Code_Generation_S_Z
-          filter: "^[s-z]"
+          name: Check_Code_Generation_O_Z
+          filter: "^[o-z]"
```
Comment (hallipr) [inline eng/pipelines/build.yml:162]: Are we intentionally skipping `n`?

Comment (archerzz) [reply eng/pipelines/build.yml]: fixed, thanks for pointing it out.

**Review by hallipr — APPROVED** [review]

**Pull request merged**